src/AV/Video.cpp

```diff
@@ -1368,10 +1368,13 @@ int Video::addTimer(int delay)
 
 bool Video::addPacket(PacketQueue *queue, AVPacket *packet)
 {
-    AVPacket *dup = av_packet_clone(packet);
-
     Packet *pkt = new Packet();
-    pkt->curr   = *dup;
+    if (packet->buf) {
```

Comment:
@verpeteren change is correct

The packets are read inside a loop here : https://github.com/nidium/Nidium/blob/8677e893fcd4a5b684d5928b6baacb2b308e0c6e/src/AV/Video.cpp#L1034-L1071 by calling `av_read_frame()` continously and adding each new (valid) packet to a queue.

- if `packet->buf != NULL` there is no need to ref it again because, because  the packet is valid indefinitely (and we are not unrefing it). 
- If `packet->buf == NULL` then we need to duplicate it (alloc + ref), because the given packet will no longer be valid on the next call to av_read_frame() (no matter if we ref it or not)

The packets added to the queue are unref() when we are retrieving packet : https://github.com/nidium/Nidium/blob/8677e893fcd4a5b684d5928b6baacb2b308e0c6e/src/AV/Video.cpp#L1250 (or when the queue is cleared)